raiden/network/transport/matrix/transport.py

```diff
@@ -361,13 +350,19 @@ def _run(self) -> None:  # type: ignore
         # run while transport parent is running
         while not self.transport._stop_event.ready():
             # once entered the critical section, block any other enqueue or notify attempt
-            with self._lock:
-                self._notify_event.clear()
-                if self._message_queue:
-                    self._idle_since = 0
-                    self._check_and_send()
-                else:
-                    self._idle_since += 1
+
+            self._notify_event.clear()
+            if self._message_queue:
+                self._idle_since = 0
+                # Shallow copy the current messages in the queue
+                # to be processed for this _check_and_send iteration
+                messages_current_workload = self._message_queue[:]
+                # then empty the message_queue
+                self._message_queue.clear()
+                with self._lock:
```

Comment:
Is this lock really necessary? `_check_and_send` is only called from `_run` and I can't see how we could end up calling `_check_and_send` on the same `RetryQueue` twice. Note that the lock is _per-RetryQueue-instance_.

Somewhat unfortunate, though, is that we manually call `_check_and_send` in `raiden/tests/mocked/test_matrix_transport.py`, but I'm hoping we could get around that.

Reply:
Yes, it is very likely unnecessary, but I wanted to be careful for now.

Reply:
And it could also be helpful to avoid later programming errors.

Reply:
The lock now was entirely removed